sunpy/coordinates/transformations.py

```diff
@@ -127,6 +132,12 @@ def hpc_to_hcc(heliopcoord, heliocframe):
     """
     Convert from Helioprojective Cartesian to Heliocentric Cartesian.
     """
+    if heliopcoord.observer != heliocframe.observer:
+        raise ConvertError("Cannot transform helioprojective coordinates to "
```

Comment:
maybe "Cannot transform directly..." since you actually can do it, you just have to go through HGS first.